<?xml version="1.0"?>
<launch>
    <arg name="node_name" default="dwa_planner_node"/>
    <arg name="path_topic" default="$(arg node_name)/path"/>
    <arg name="twist_stamped_topic" default="$(arg node_name)/twist_stamped"/>
    <arg name="twist_cmd_topic" default="$(arg node_name)/target_twist"/>
    <arg name="current_pose_topic" default="$(arg node_name)/current_pose"/>
    <arg name="grid_map_topic" default="$(arg node_name)/grid_map"/>
    <arg name="robot_frame" default="base_link"/>
    <arg name="layer_name" default="base_layer"/>

    <node pkg="dwa_planner" type="dwa_planner_node" name="$(arg node_name)" output="screen" respawn="false" respawn_delay="0">
        <param name="path_topic" value="$(arg path_topic)"/>
        <param name="twist_stamped_topic" value="$(arg twist_stamped_topic)"/>
        <param name="twist_cmd_topic" value="$(arg twist_cmd_topic)"/>
        <param name="current_pose_topic" value="$(arg current_pose_topic)"/>
        <param name="robot_frame" value="$(arg robot_frame)"/>
        <param name="grid_map_topic" value="$(arg grid_map_topic)"/>
        <param name="layer_name" value="$(arg layer_name)"/>
    </node>
</launch>